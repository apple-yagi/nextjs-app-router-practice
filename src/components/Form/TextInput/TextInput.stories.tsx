import { Meta, StoryObj } from "@storybook/react";
import { within } from "@storybook/testing-library";

import { TextInput } from ".";

const meta = {
  component: TextInput,
} satisfies Meta<typeof TextInput>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
  args: {
    label: "Label",
  },
} satisfies Story;

export const ClickedLabel = {
  ...Default,
  play: async ({ canvasElement }) => {
    const ui = within(canvasElement);
    await ui.getByLabelText("Label").focus();
  },
} satisfies Story;
