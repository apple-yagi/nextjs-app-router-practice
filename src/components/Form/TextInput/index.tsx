import { ComponentPropsWithoutRef } from "react";
import { twMerge } from "tailwind-merge";

type Props = {
  label: string;
} & ComponentPropsWithoutRef<"input">;

export const TextInput = ({ className, label, ...props }: Props) => {
  return (
    <div className="flex flex-col gap-1">
      <label htmlFor={label}>{label}</label>
      <input
        className={twMerge(
          "block w-full rounded-lg border border-gray-300 bg-gray-100 p-4 focus:border-primary",
          className
        )}
        id={label}
        type="text"
        {...props}
      />
    </div>
  );
};
