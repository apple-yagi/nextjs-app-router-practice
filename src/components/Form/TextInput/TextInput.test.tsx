import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import * as stories from "./TextInput.stories";

const { Default, ClickedLabel } = composeStories(stories);

describe("TextInput", () => {
  it("正しく表示されること", () => {
    expect(render(<Default />).asFragment()).toMatchSnapshot();
  });

  it("Labelをクリックした時にフォーカスされること", async () => {
    const { container, getByRole } = render(<ClickedLabel />);

    await ClickedLabel.play({ canvasElement: container });

    expect(getByRole("textbox")).toHaveFocus();
  });
});
