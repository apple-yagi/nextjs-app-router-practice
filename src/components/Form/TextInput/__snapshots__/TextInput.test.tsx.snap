// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextInput > 正しく表示されること 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
  >
    <label
      for="Label"
    >
      Label
    </label>
    <input
      class="block w-full rounded-lg border border-gray-300 bg-gray-100 p-4 focus:border-primary"
      id="Label"
      type="text"
    />
  </div>
</DocumentFragment>
`;
